### PR TITLE
fix(components/VirtualizedList): hide key/value pairs when there is no value

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
I want to hide key/value pair when the value does not exist (only for expanded list view).
But, for now, expanded list view presents non-natural order which can produce issue when you want to hide one or many cells.

image 10

**What is the chosen solution to this problem?**
Fix flow order and be able to hide cell when the value does not exist.

image

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by annilsson** [comment]
This is different from the guidelines images so we should also update the guidelines if we do this? http://guidelines.talend.com/document/92132#/navigation-layout/list

**Comment by frassinier** [comment]
@annilsson order is not defined in the guidelines? 
Summon @tlnd-clalande, @AnLefevre and @mhuet 🙏

File: packages/components/src/VirtualizedList/RowLarge/RowLarge.component.js [hunk 1]
```diff
@@ -28,19 +29,25 @@ class RowLarge extends React.Component {
...
-				</dt>
-				<dd className={theme['field-value']} title={tooltip}>
-					{cellContent}
-				</dd>
-			</div>
-		);
+		const rawContent = getRowData(parent, index);
+		const dataKey = getDataKey(field);
+		const value = rawContent[dataKey];
+		if (value != null && `${value}`.trim().length) {
```
Comment (jdescottes-talend) [inline packages/components/src/VirtualizedList/RowLarge/RowLarge.component.js]: Why checking if need to display based on `rawContent` when its `cellContent` which is used ?

Comment (frassinier) [reply packages/components/src/VirtualizedList/RowLarge/RowLarge.component.js]: In my case, `cellContent` contains extra div with empty content.
I can't lead on.

Comment (jdescottes-talend) [reply packages/components/src/VirtualizedList/RowLarge/RowLarge.component.js]: yep nevermind, I didn't chek the apis ;)

**Comment by tlnd-clalande** [comment]
@frassinier (checked with @mhuet)
The order is Left to the right and back to line (like your 2nd screen).

Question ? is ne number of "column" is responsive ? or is it only 3 ?

**Comment by frassinier** [comment]
@tlnd-clalande 
>The order is Left to the right and back to line (like your 2nd screen).
Thanks! 🎉 

>Question ? is ne number of "column" is responsive ? or is it only 3 ?
Only 3 at the moment, nothing was changed

File: packages/components/src/VirtualizedList/RowLarge/RowLarge.component.js [hunk 2]
```diff
@@ -28,19 +29,25 @@ class RowLarge extends React.Component {
...
+					<dt key={fieldIndex} className={theme['field-label']}>
+						{label}
+					</dt>
+					<dd className={theme['field-value']} title={tooltip}>
+						{cellContent}
+					</dd>
+				</div>
+			);
+		}
+		return null;
```
Comment (jmfrancois) [inline packages/components/src/VirtualizedList/RowLarge/RowLarge.component.js]: please escape as soon as possible and finish by returning the JSX

**Comment by frassinier** [comment]
👋 @jdescottes-talend and @jmfrancois 

I took your change requests into account

**Pull request merged**